doc/source/user/quickstart.rst

```diff
@@ -881,7 +881,7 @@ creates a new array object that looks at the same data.
     >>> c.flags.owndata
     False
     >>>
-    >>> c = c.reshape((2, 6))  # a's shape doesn't change
+    >>> c = c.reshape((2, 6))  # a's shape doesn't change, reassigned c is still a view of a
```

Comment:
I got triggered by the reassignment of `c`  since `c` was initially assigned a view of `a`, i.e.
```python
c = a.view()
```
then later:
```python
c = c.reshape(()2, 6)
```
Since the comment mentions "no change" and the statement has a reassignment I thought it needed some clarification. Explicitly mentioning that `c` is reassigned also communicates that the `reshape()` method returns an array and does not operate in place.

Reply:
I think the paradigm of re-using a variable name is confusing. Could you do something like `ctag = c.reshape(...`

Reply:
I agree that re-using a variable name might be confusing. The take home message of the line in question is that changing the shape of the view `c` on `a` does not change `a`. Maybe it would be even better to use a *in place operation* instead of a method returns an array, i.e.:

```python
c.shape = (2, 6)  # a's shape doesn't change
```

or:
```python
c.resize((2, 6))  # a's shape doesn't change
```

If the above is okay, any preference for any of the two or include both?

Reply:
> `c.shape = (2, 6)`

I would prefer to deprecate this syntax, since is truely modifies the underlying ndarray, is not thread safe, and does not always succeed. It depends on refcount semantics to make sure there is only one instance of `c` in the interpreter, which doesn't always work.

What about `b = a.resize((2, 6))`, and mention that it too creates a new view on `a`, without changing `a`'s shape?

Reply:
`b` is already been taken by the previous example:

```python
b = a            # no new object is created
b is a           # a and b are two names for the same ndarray object
```

Using ```resize()``` on `c` uses the ```ndarray.resize()``` method which operates in place, it is not the ```numpy.resize()``` function which returns a new array. ```id(c)``` is the same before and after calling ```c.resize((2, 6))```. Therefore, in this case, the same objection holds for ```resize()``` as ```c.shape```. Moreover, ```b = a.resize((2, 6))``` assigns ```None``` to `b`.

Given the context, we need to find a solution that:

- Clearly demonstrates that modifying a view doesn't affect the original array
- Avoids reassignment, which was the original point of confusion
- Addresses mattip's concerns about thread safety and refcount semantics

Considering all points raised in this discussion, I propose the following solution:

```python
c = a.view()
c.shape = (2, 6)  # modifies c in-place, a's shape doesn't change
```

Rationale:

1. This uses ```c.shape = (2, 6)``` which modifies `c` in-place without reassignment, avoiding the original confusion about ```c = c.reshape((2, 6))```.
2. We avoid using ```resize()``` and explaining the difference between ```ndarray.resize()``` and ```numpy.resize()``` here to keep the example simple and focus on the core concept of view behavior.
3. It clearly demonstrates that modifying the view `c` doesn't affect the original array `a`.
4. While `c.shape = (2, 6)` has some drawbacks as mattip mentioned (not thread-safe, depends on refcount semantics), it's still a commonly used and understood NumPy operation.

To address mattip's concerns, we could add a note in the documentation:
Note: While `c.shape = (2, 6)` is used here for clarity, it's important to be aware that this operation
directly modifies the underlying array and is not thread-safe. In production code, especially in 
multi-threaded environments, it's often safer to use methods like `reshape()` which return new views.

Reply:
I think the discussion is trending towards a much greater level of detail than was perhaps originally intended; in the interest of making sure this isn't stuck, I'm just going to go ahead and merge (my comment wasn't a blocker anyways).

If we want to keep iterating let's go ahead and do so in a follow-up PR. Thanks @kplanken !

Reply:
I agree, thank you for the merge and keeping us from derailing :-)